pkg/providers/imagefamily/nodeimageversionsclient.go

```diff
@@ -102,21 +104,29 @@ func FilteredNodeImages(nodeImageVersions []NodeImageVersion) []NodeImageVersion
 	return filteredImages
 }
 
+// isNewerVersion will return if version1 is greater than version2, note the new versioning scheme is yearmm.dd.build, previously it was yy.mm.dd without the build id.
```

Comment:
Q: Should this code try to solve the windows versioning paradigm? Windows vhd versions are very different, and we don't support it yet. 

AKS Windows VHD versions use a different versioning scheme, typically in the format Build.Revision.YYMMDD. For example, a version like 17763.6532.241114 can be broken down as follows:
	•	Build Number: 17763
	•	Revision Number: 6532
	•	Date Code: 241114 (interpreted as November 14, 2024)
	
Should we make sure this algorithm also supports this type of version id? or leave that as a problem to solve for windows support?